talisker/logs.py

```diff
@@ -318,32 +325,41 @@ def remove_quotes(self, s):
 
     def logfmt(self, structured):
         logfmt = (self.logfmt_atom(*kv) for kv in structured.items())
-        return " ".join(logfmt)
+        return " ".join(l for l in logfmt if l is not None)
 
     def logfmt_atom(self, k, v):
-        # we need unicode strings so as to be able to replace
-        if isinstance(k, bytes):
-            k = k.decode('utf8')
+        # we need unicode strings so as to be able to do replace
+        if not v:
+            return None
+
         if not isinstance(v, str):
             if isinstance(v, bytes):
                 v = v.decode('utf8')
             elif not isinstance(v, str):
```

Comment:
Isn't this just "else", given it's already inside an if block with the same predicate?